runtime/bcutil/cfreader.c

```diff
@@ -1782,10 +1782,14 @@ checkMethods(J9PortLibrary* portLib, J9CfrClassFile* classfile, U_8* segment, U_
 		if (nameIndexOK && utf8Equal(&classfile->constantPool[method->nameIndex], "<init>", 6)) {
 
 			/* check no invalid flags set */
-			if (value & ~CFR_INIT_METHOD_ACCESS_MASK) {
-				errorCode = J9NLS_CFR_ERR_INIT_METHOD__ID;
-				goto _errorFound;
-			}
+			if ((value & ~CFR_INIT_METHOD_ACCESS_MASK)
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+				&& !J9_ARE_ALL_BITS_SET(classfile->accessFlags, CFR_ACC_VALUE_TYPE)
+#endif /* defined(J9VM_OPT_VALHALLA_VALUE_TYPES) */
+				) {
```

Comment:
problem with indentation here

Reply:
actually instead doing this, you can redefine `CFR_INIT_METHOD_ACCESS_MASK` for VTs in cfr.h.

`CFR_ACC_PUBLIC | CFR_ACC_PRIVATE | CFR_ACC_STATIC | CFR_ACC_SYNTHETIC | CFR_ACC_VARARGS | CFR_ACC_STRICT`

Reply:
Hmm the macro doesn't work. I think it's because the class has CFR_ACC_VALUE_TYPE but the `value` doesn't. `value` represents the method's access flags, not hte classfile's. Do you think we should make the method have that flag?

Reply:
What I meant was if its a VT do this check instead `(value & ~CFR_VT_INIT_METHOD_ACCESS_MASK)`